### PR TITLE
feat(android): generate kotlin files at build time

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [x] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: src/webview/android/kotlin/TauriActivity.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package {{app-domain-reversed}}.{{app-name-snake-case}}
```
Comment (lucasfernog) [inline src/webview/android/kotlin/TauriActivity.kt:1]: should this one be generated by tao?

Comment (amrbashir) [reply src/webview/android/kotlin/TauriActivity.kt:1]: iirc, tao generates functions using `android_binding!` macro, which takes the app_domain and app_name as input.

Comment (lucasfernog) [reply src/webview/android/kotlin/TauriActivity.kt:1]: Yeah but the TauriActivity stuff is still related to the tao code, so in this setup if tao changes, we'd need to change wry too.

File: build.rs [hunk 2]
```diff
@@ -9,4 +9,37 @@ fn main() {
...
     println!("cargo:rustc-link-lib=framework=WebKit");
   }
+
+  let is_android = std::env::var("TARGET")
+    .map(|t| t.ends_with("-android"))
+    .unwrap_or_default();
+  if is_android {
+    use std::{fs, path::PathBuf};
+
+    fn env_var(var: &str) -> String {
```
Comment (lucasfernog) [inline build.rs:19]: we need to add a `println!("cargo:rerun-if-env-changed={var}");` here

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**